templates/_helpers.tpl

```diff
@@ -191,6 +191,23 @@ Reduce duplication of the st2.*.conf volume details
 {{- end }}
 {{- end -}}
 
+{{- define "stackstorm-ha.init-containers-wait-for-auth" -}}
+- name: wait-for-auth
+  image: "alpine/curl:8.4.0"
+  imagePullPolicy: {{ .Values.image.pullPolicy }}
+  command:
+    - 'sh'
+    - '-c'
+    - >
+      until curl -skSL --fail -w '\n'  -X POST -u {{ .Values.st2.username }}:{{ .Values.st2.password }} "https://{{ required ".Values.ingress.fqdn is required if .Values.ingress.class is non-empty" .Values.ingress.fqdn | printf (ternary "canary-%s" "%s" .Values.phaseCanary)}}/auth/tokens"; do
```

Comment:
I don't see these vars in our values file:
- `ingress.fqdn`
- `ingress.class`
- `phaseCanary`

https://github.com/StackStorm/stackstorm-k8s/blob/17e5fca27a38164d77160e341ce8ba8d531f50e0/values.yaml#L298-L321

I also do not use the ingress, so these would never be defined for me.

user/pass is also only available in my cluster(s) during initial bootstrap. Once I switch to ldap, I exclusively use api tokens within helm.

Do you have any ideas on how to make this more generic?